### PR TITLE
Add void whitelist item for switch expression case

### PR DESCRIPTION
**Pull request by eernstg**
Cf. https://github.com/dart-lang/language/issues/2907.

### CODE REVIEW

File: accepted/future-releases/0546-patterns/feature-specification.md [hunk 1]
```diff
@@ -2158,6 +2158,19 @@ issue an error since the empty placeholder switch is no longer exhaustive,
 allowing the user to find all the places in the code that need to be updated to
 handle the new subtype.*
 
+We add the following item to the list of [locations where it is not an error
+to have an expression of type `void`][void whitelist]:
```
Comment (lrhn) [inline accepted/future-releases/0546-patterns/feature-specification.md:2162]: whitelist -> allowlist

File: accepted/future-releases/0546-patterns/feature-specification.md [hunk 2]
```diff
@@ -2158,6 +2158,19 @@ issue an error since the empty placeholder switch is no longer exhaustive,
...
+
+[void whitelist]:https://github.com/dart-lang/language/blob/f58f8c5d24eed253db41474a3149f7b79a274c63/specification/dartLangSpec.tex#L21981
+
+- In a `<switchExpressionCase>` of the form `p => e`, `e` may have type `void`.
+
+*This means that we treat switch expressions similarly to conditional
+expressions with respect to the type `void`. When one or more cases in a switch
+expression have type `void`, the switch expression as a whole will have type
+`void`, which means that it can in turn only occur in a whitelisted location.
```
Comment (lrhn) [inline accepted/future-releases/0546-patterns/feature-specification.md:2171]: ... in an allowed location

File: accepted/future-releases/0546-patterns/feature-specification.md (hunk 1 above)
Comment (eernstg) [reply accepted/future-releases/0546-patterns/feature-specification.md:2162]: Done.

File: accepted/future-releases/0546-patterns/feature-specification.md (hunk 2 above)
Comment (eernstg) [reply accepted/future-releases/0546-patterns/feature-specification.md:2171]: Done.

**Review by munificent — APPROVED** [review]
👍

**Pull request merged**